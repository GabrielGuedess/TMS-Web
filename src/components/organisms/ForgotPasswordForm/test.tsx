import { render } from '@testing-library/react';

import { ForgotPasswordForm } from '.';

const push = jest.fn();

jest.mock('next/navigation', () => ({
  useRouter: () => ({
    push,
  }),
}));

describe('<ForgotPasswordForm />', () => {
  it('should render the forgot password form correctly', () => {
    const { container } = render(<ForgotPasswordForm />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
