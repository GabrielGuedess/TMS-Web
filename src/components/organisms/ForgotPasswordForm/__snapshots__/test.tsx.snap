// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgotPasswordForm /> should render the forgot password form correctly 1`] = `
<form>
  <div
    class="flex flex-col gap-8"
  >
    <div
      class="relative flex h-min flex-col"
    >
      <div
        class="relative"
      >
        <input
          autocomplete="on"
          class="peer w-full rounded-lg border-2 bg-transparent font-normal text-comet-500 outline-none transition-all placeholder:opacity-0 dark:border-shark-950 dark:text-white-lilac-50 p-3 focus:border-primary-300 dark:focus:border-primary-400 focus:placeholder:opacity-100"
          id=":r0:"
          name="email"
          required=""
          type="email"
        />
        <label
          class="pointer-events-none absolute left-4 top-1/2 z-10 -translate-y-1/2 px-[3px] text-sm text-comet-500 transition-all before:absolute before:inset-0 before:-z-10 before:block before:w-full before:bg-white-lilac-50 before:transition-all peer-valid:top-[-35%] peer-valid:translate-y-1/2 peer-read-only:top-[-35%] peer-read-only:translate-y-1/2 peer-focus:top-[-35%] peer-focus:translate-y-1/2 dark:text-dark-300 before:dark:bg-smoke-950 peer-focus:text-primary-400"
          for=":r0:"
        >
          E-mail
        </label>
      </div>
    </div>
    <div
      class="flex flex-col gap-6"
    >
      <button
        class="relative transition-transform duration-200 ease-in-out w-full hover:scale-105 focus:scale-105 active:scale-90 h-12"
        type="submit"
      >
        <div
          aria-label="Button Background"
          class="absolute size-full rounded-md opacity-[16%] bg-primary-400"
        />
        <div
          class="flex h-full items-center justify-center gap-2 px-5 py-[0.625rem] flex-row text-primary-400"
        >
          <span
            class="z-10 font-medium leading-[1.125rem] tracking-[0.027rem] text-[0.9375rem]"
          >
            Enviar Solicitação
          </span>
        </div>
      </button>
      <a
        class="flex h-12 w-full items-center justify-center text-sm font-semibold text-shark-950 outline-primary-400 transition-all hover:text-primary-400 focus:text-primary-400 dark:text-white-lilac-50 dark:hover:text-primary-400 dark:focus:text-primary-400"
        href="/"
      >
        <svg
          height="16"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13.83 19a1 1 0 0 1-.78-.37l-4.83-6a1 1 0 0 1 0-1.27l5-6a1 1 0 0 1 1.54 1.28L10.29 12l4.32 5.36a1 1 0 0 1-.78 1.64"
            fill="currentColor"
          />
        </svg>
        Voltar para login
      </a>
    </div>
  </div>
</form>
`;
